lib/Command/ExpirePassword.php

```diff
@@ -106,54 +130,96 @@ protected function configure() {
 	 * @throws \OCP\PreConditionNotMetException
 	 */
 	protected function execute(InputInterface $input, OutputInterface $output) {
-		$uid = $input->getArgument('uid');
+		$groups = $input->getOption('group');
+		$allUsers = $input->getOption('all');
+		$uids = $input->getOption('uid');
 
-		/** @var $user \OCP\IUser */
-		$user = $this->userManager->get($uid);
-
-		if ($user === null) {
-			$output->writeln("<error>Unknown user: $uid</error>");
-			return self::EX_NOUSER;
+		$expireDate = new \DateTime();
+		$oldDate = new \DateTime();
+		$timeZone = new \DateTimeZone('UTC');
+		$inputExpireDate = $input->getArgument('expiredate');
+
+		if ($allUsers !== false) {
+			$this->userManager->callForAllUsers(function (IUser $user) use ($expireDate, $oldDate, $timeZone, $inputExpireDate, $output) {
+				$calculatedExpireDate = $this->setPasswordExpiry($expireDate, $oldDate, $timeZone, $inputExpireDate, $user);
+				$output->writeln('The password for ' . $user->getUID() . ' is set to expire on ' . $calculatedExpireDate->format('Y-m-d H:i:s T') . '.');
+			});
+		} elseif (\count($groups) >= 1) {
+			$users = [];
+
+			foreach ($groups as $group) {
+				if ($this->groupManager->groupExists($group) === true) {
+					foreach ($this->groupManager->findUsersInGroup($group) as $user) {
+						if (!\in_array($user->getUID(), $users)) {
+							$calculatedExpireDate = $this->setPasswordExpiry($expireDate, $oldDate, $timeZone, $inputExpireDate, $user);
+							$output->writeln('The password for ' . $user->getUID() . ' is set to expire on ' . $calculatedExpireDate->format('Y-m-d H:i:s T') . '.');
+							$users[] = $user->getUID();
```

Comment:
It's better to store the user ids as the key. We expect the user to not be in the array, let's say the 90% of the times (having the user in several of these groups shouldn't happen too often), so most of the times you'll need to traverse the whole array over and over (80-90% of the times), and the array will increase.

Storing the uids as keys will allow us to check the existence of the key via `isset($users[$uid])`, which will perform the check faster and in constant time regardless of the size of the array. You might want to store it like `$users[$user->getUID()] = true` for convenience. I don't think we'll need the user object later.